File: src/main/java/br/com/controllers/Usuario/ClienteController.java
package br.com.controllers.usuario;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;
import br.com.services.classes.usuario.ClienteService;
import br.com.model.entities.classes.usuario.Cliente;

@RestController
@CrossOrigin(origins = "*")
public class ClienteController extends ClienteService {

    private final String URL = "/cliente";

    @GetMapping(value = URL)
    protected ResponseEntity<Iterable<Cliente>> procurarTodos() {
        return super.procurarTodos();
    }

    @PostMapping(value = URL)
    public ResponseEntity<Cliente> salvar(@RequestBody Cliente cliente) throws Exception {
        return super.salvar(cliente);
    }

    @DeleteMapping(URL)
    public ResponseEntity<String> remover(@RequestBody Cliente cliente) throws Exception {
        return super.remover(cliente);
    }

    @PatchMapping(URL)
    public ResponseEntity<Cliente> atualizar(@RequestBody Cliente cliente) throws Exception {
        return super.atualizar(cliente);
    }

    @GetMapping(URL + "/{id}")
    public ResponseEntity<Cliente> procurarPorID(@PathVariable Integer id) {
        return super.procurarPorID(id);
    }

    @GetMapping(URL + "/Buscar")
    public ResponseEntity<Cliente> buscarPorCnpjCpf(@RequestBody Cliente cliente) {
        return super.buscarPorCnpjCpf(cliente.getCnpjCpf());
    }

    @Override
    public void validate(Cliente entity) throws Exception {
        if (entity.getEmail() == null || entity.getEmail().isEmpty()) {
            throw new Exception("\n\nEmail não pode ser vazio!\n\n");
        }

        if (entity.getNome() == null || entity.getNome().isEmpty()) {
            throw new Exception("\n\nNome não pode ser vazio!\n\n");
        }
    }
}

File: src/main/java/br/com/controllers/usuario/ClienteController.kt
package br.com.controllers.usuario

import br.com.model.usuario.Cliente
import br.com.services.classes.usuario.ClienteService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@CrossOrigin(origins = ["*"])
class ClienteController(@Autowired private val clienteService: ClienteService) {

    companion object {
        private const val URL = "/cliente"
    }

    @GetMapping(value = [URL])
    fun procurarTodos(): ResponseEntity<Iterable<Cliente>> {
        return clienteService.procurarTodos()
    }

    @PostMapping(value = [URL])
    @Throws(Exception::class)
    fun salvar(@RequestBody cliente: Cliente): ResponseEntity<Cliente> {
        return clienteService.salvar(cliente)
    }

    @DeleteMapping(URL)
    @Throws(Exception::class)
    fun remover(@RequestBody cliente: Cliente): ResponseEntity<String> {
        return clienteService.remover(cliente)
    }

    @PatchMapping(URL)
    @Throws(Exception::class)
    fun atualizar(@RequestBody cliente: Cliente): ResponseEntity<Cliente> {
        return clienteService.atualizar(cliente)
    }

    @GetMapping("$URL/{id}")
    fun procurarPorID(@PathVariable id: Int?): ResponseEntity<Cliente> {
        return clienteService.procurarPorID(id!!)
    }

    @GetMapping("$URL/Buscar")
    fun buscarPorCnpjCpf(@RequestBody cliente: Cliente): ResponseEntity<Cliente> {
        return clienteService.buscarPorCnpjCpf(cliente.cnpjCpf)
    }

    @Throws(Exception::class)
    fun validate(entity: Cliente) {
        if (entity.email.isBlank()) {
            throw Exception("\n\nEmail não pode ser vazio!\n\n")
        }
        if (entity.nome.isBlank()) {
            throw Exception("\n\nNome não pode ser vazio!\n\n")
        }
    }
}
